undertow/WFLY-10614_Elytron_SecurityDomain_Referencing.adoc

```diff
@@ -0,0 +1,98 @@
+
+Other types of deployment will make use of the authentication APIs provided by the Servlet specification as an alternative to using authentication mechanisms so for these the factory is redundant.  Deployments could also make use of the WildFly Elytron SecurityDomain associated with the deployment directly again bypassing the need for authentication mechanisms and the related factories.
+
+Finally there is another type of deployment where some other library is handling the authentication of the deployment and the only step needed is to associate the results with a SecurityIdentity from the current SecurityDomain, a couple of examples here are the Apache CXF WSS4J integration and the upcoming WildFly Elytron JASPIC implementation, in both of these cases the verification can be performed completely independently of the WildFly Elytron security solution but we still need to be able to create an identity from the current SecurityDomain and use it for the current request.
+
+Overall whilst referencing a `http-authentication-factory` provides a lot of flexibility for advanced configuration there are situations where this leads to configuration that is more complex than is really required and other situations where an authentication factory is referenced as that is the only way to associate a WildFly Elytron SecurityDomain with a web application deploynment.  This proposal is to update the application-security-domain resource in the Undertow subsystem so that instead of referencing a http-authentication-factory a security domain can be referenced instead.
+ 
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10614[WFLY-10614]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1024[EAP7-1024]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+* WildFly Core
+** Default WildFly Elytron Configuration
+
+* WildFly
+** Undertow Subsystem
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+The `application-security-domain` resource in the Undertow subsystem will support a new `security-domain` attribute as an alternative to the existing `http-authentication-factory` attribute - these two attributes will be defined as 'alternatives' to ensure mutual-exclusivity.
+
+Where a deployment matches against an `application-security-domain` mapping, if that mapping references a `security-domain` then provided the web.xml is configured to use the mechanisms referenced in the servlet specification (BASIC, CLIENT_CERT, DIGEST, or FORM) then these will still be enabled for the application without requiring a `http-authentication-factory` reference.
```

Comment:
In this case Elytron mechanism implementation will be used or Undertow implementation?

Reply:
That is correct, this enhancement is predominantly a configuration simplification to cut our a part of the chain of configuration that would be redundant for many users.